slf4j-api/src/main/java/org/slf4j/MDC.java

```diff
@@ -156,6 +196,40 @@ public static MDCCloseable putCloseable(String key, String val) throws IllegalAr
         return new MDCCloseable(key);
     }
 
+    /**
+     * Put a diagnostic context map (the <code>entries</code> parameter) as identified with each
+     * <code>key</code> into the current thread's diagnostic context map. The
+     * <code>entries</code> parameter and its <code>keys</code> cannot be null. The
+     * <code>value</code> of each entry can be null only if the underlying implementation
+     * supports it.
+     *
+     * <p>
+     * This method delegates all work to the MDC of the underlying logging system.
+     * <p>
+     * This method returns a <code>Closeable</code> object which can remove all <code>keys</code>
+     * that are part of the <code>entries</code> when <code>close</code> is called.
+     *
+     * <p>
+     * Useful with Java 9 for example :
+     * <code>
+     *   try(MDC.MDCCloseable closeable = MDC.putAllCloseable(Map.of(key1, value, key2, value2)) {
+     *     ....
+     *   }
+     * </code>
+     *
+     * @param entries entries to put in the map
+     * @return a <code>Closeable</code> who can remove all originally supplied <code>keys</code>
+     * when <code>close</code> is called.
+     *
+     * @throws IllegalArgumentException
+     *           in case the "entries" parameter is null
+     * @since 2.0.0
+     */
+    public static MDCCloseable putAllCloseable(Map<String, String> entries) throws IllegalArgumentException {
```

Comment:
Would an override like `java.util.Map.of(String, String, ...)` be useful here?

In many of the places in applications where I have wanted to use this sort of functionality, I have usually wanted to add several entries at once. In this case it would read something like

```java
try (MDC.putAllCloseable(
    "X-Request-ID", request.getHeader("X-Request-ID"),
    "User-Agent", request.getHeader("User-Agent"),
    "From", request.getHeader("From")
)) {
    processRequest();
}
```
versus having to declare this separately.